specs/lazy-adr/adr-011-based-sequencing.md

```diff
@@ -0,0 +1,188 @@
+- ExecuteTxs(txs): given the batch of transactions, execute them using the execution layer to obtain the updated state root.
+- SetFinal(block): after the batches are validated (inclusion and state update confirmed), instruct the execution layer to mark as finalized.
+
+## Alternative Approaches
+
+There are two approaches to implementing the based sequencing rollups:
+- No proposer, every node creates batches by pulling the base layer transactions (included and sequenced) by following the global forkchoice rule (e.g., max bytes supported by the execution layer).
+- a proposer can be utilized for efficiency purposes without introducing additional security assumptions where the proposer front-runs the batch preparation and provides (in the p2p layer) the prepared batch along with the updated state and proofs of inclusion and state update.
+
+## Decision
+
+A based sovereign evm rollup using Rollkit.
+
+Components:
+- Celestia for based sequencing
+- Rollkit for consensus
+- EVM for execution (reth and op-geth to allow sequenced transaction submission)
+
+We decide to use the proposer based approach for efficiency reason as well as utilizing the single sequencer implementation that already exists with minimal configurational changes.
+
+## Detailed Design
+
+### Sequence diagram
+
+The sequence digram below demostrates the interactions between the user, the execution layer, the sequencing layer, and the consensus layer (rollkit's block manager). In the no-proposer mode, every node (propose or full node) will follow all the interactions, whereas in case of proposer mode, the full nodes will follow all the interactions, however, they p2p layer front runs the sharing of the batch (created by the proposer) and state updates (in the form of headers).
+
+```mermaid
+sequenceDiagram
+    participant User as user
+    participant Reth as reth
+    participant Exec as go-execution-evm
+    participant Reaper as reaper
+    participant Header as header proposer
+    participant BlockMgr as block manager
+    participant Sequencer as based-sequencer
+    participant Base as base layer
+
+    User->>Reth: submits rollup tx
+
+    Reaper->>Exec: GetTxs
+    Exec->>Reth: get txs from mempool
+    Reaper->>Sequencer: SubmitRollupTxs
+    Sequencer->>Base: Submit blobs
+
+    BlockMgr->>Sequencer: GetNextBatch
+    Sequencer->>Base: Retrieve blobs
+    Sequencer->>BlockMgr: batch w/ proofs
+
+    BlockMgr->>Exec: ExecuteTxs
+    Exec->>Reth: call engine api to create new evm block
+
+    Header->>Exec: SetFinal
+    Exec->>Reth: call engine api to finalize the state
+
+    Reth-->>User: user can query tx being final
+```
+
+### Transaction submission
+
+- User can directly submit the rollup transaction to base layer. User need to compose a base layer transaction using the signed bytes of the rollup transaction. User also need to have access to base layer light node and an account for paying the base layer fee for transaction submission.
```

Comment:
could we do a diagram on this flow and the below flow to help show the difference. the above is option 1 but option two is missing